src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -1213,6 +1231,93 @@ private void ShowGeneralSettingsOptionPage(object sender, EventArgs args)
             ShowOptionPageSafe(typeof(GeneralOptionPage));
         }
 
+        private void ExecuteClearNuGetLocalResourcesCommand(object sender, EventArgs e)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                try
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    var isUserContinuing = MessageHelper.ShowQueryMessage(
+                        message: Resx.VSOptions_Text_ClearLocalsPromptMessage,
+                        title: Resx.VSOptions_Text_ClearLocalsPromptTitle,
+                        showCancelButton: false,
+                        defaultButton: OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_SECOND);
+
+                    NavigatedTelemetryEvent evt;
+
+                    if (isUserContinuing == true)
+                    {
+                        var clearNuGetLocalsViewModel = new ClearNuGetLocalsViewModel(ClearNuGetLocalsCommandExecute);
+                        OutputConsoleLogger.Value.Start();
+                        var clearNuGetLocalResourcesWindow = new ClearNuGetLocalResourcesWindow(clearNuGetLocalsViewModel);
+                        clearNuGetLocalResourcesWindow.ShowModal();
+                        evt = NavigatedTelemetryEvent.CreateWithClearLocalsCommand(isUnifiedSettings: true, isPromptCancelled: false);
+                    }
+                    else
+                    {
+                        evt = NavigatedTelemetryEvent.CreateWithClearLocalsCommand(isUnifiedSettings: true, isPromptCancelled: true);
+                    }
+                    TelemetryActivity.EmitTelemetryEvent(evt);
+                }
+                catch (Exception ex)
+                {
+                    LogError(ex.Message);
+                    ActivityLog.LogError(NuGetUI.LogEntrySource, ex.ToString());
+                    throw ex;
+                }
+                finally
+                {
+                    OutputConsoleLogger.Value.End();
+                }
+            }).PostOnFailure(nameof(NuGetPackage), nameof(ExecuteClearNuGetLocalResourcesCommand));
+        }
+
+        public async Task ClearNuGetLocalsCommandExecute()
```

Comment:
The usual recommendation is to name all Task-returning methods with an "Async" suffix (and accept a cancellation token)